### ISSUE TITLE
Jet hints fail on copies of the standard library

### ISSUE THREAD

**Issue by belisarius222** [urbit/urbit#1100]
Using Ford-turbo's organic dairy linking, you can import libraries and then `=,` them to dump their inner namespaces into the subject. This will often expose a (logical) copy of the standard library into the subject, since libraries are often cores whose context contains the standard library.

As it is now, these copies of the standard library can't be used in the chain of parent cores for jet hints. It's unclear why, but it's probably a bug.

Here's an example of using `..^is` as the parent core instead of `..is` to work around the problem:
```
/+  some-lib
=,  some-lib
::
~%  %some-core  ..^is  ~
|%  ++  foo  ~  --
```

**Comment by belisarius222** [comment]
@frodwith

**Comment by cgyarvin** [comment]
My strongly held view is that libraries should always be %lead — they should never expose their payloads.  Thus, ^add should never work. 

Change my mind!  But why would we want these unintended exports cluttering up our namespace?

**Comment by frodwith** [comment]
We don't, and there should never be a ..^is. But I don't see any reason why this registration shouldn't work, anyway.

**Comment by joemfb** [comment]
I restored the top-level arvo gate in urbit/arvo#900, and also tweaked the %ford %reef build to unwrap `+ride`. This fixed jet registration errors in `+disabled-test-reef-slow`, but maybe something else was depending on the old behavior?

**Comment by belisarius222** [comment]
This was broken before that. In any case, all the copies of the standard
library are just Ford's :pit, which it passes through unchanged from Arvo
unless you do something weird. This means incompatibilities in a standard
library that Ford created wouldn't show up here, since it doesn't get
called during typical operation.

—
~rovnys-ricfer
https://urbit.org

**Comment by zalberico** [comment]
Looking at this and [#932 ](https://github.com/urbit/arvo/pull/932) - it's not clear (to me) if there's still some specific thing that should come out of this. @belisarius222 @joemfb thoughts?

**Comment by belisarius222** [comment]
I think this is obsolete.

—
~rovnys-ricfer
https://tlon.io

**Comment by joemfb** [comment]
I don't think it is -- there are correctness questions for the jet dashboard, and requirements questions for userspace imports (ie, should `/+` and `/-` always `^?). The latter is definitely still live, there's not demonstration here that the former is not too.
